api/v1/hostdefinition_types.go

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2022 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// HostDefinitionSpec defines the desired state of HostDefinition
+type HostDefinitionSpec struct {
+	HostDefinition IBMBlockCSIHostDefinitionSpec `json:"hostDefinition"`
+
+	ImagePullSecrets []string `json:"imagePullSecrets,omitempty"`
+}
+
+// IBMBlockCSIHostDefinitionSpec defines the observed state of HostDefinition
+type IBMBlockCSIHostDefinitionSpec struct {
+	Repository string `json:"repository"`
+	Tag        string `json:"tag"`
+
+	// +kubebuilder:validation:Optional
+	Prefix string `json:"prefix"`
+
+	// +kubebuilder:validation:Optional
```

Comment:
do we want to have it optional?
do we want to take the risk of defining it on a connectivity type he didn't intend to?

Reply:
I think it's ok if the user won't specify we will create the hosts with connectivity by the nodeID, we will write it in the documentation and in phase 2 we will do it in more "smart" way, but do you think that we should put it as mandatory? @ArbelNathan @oriyarde @erantzabari

Reply:
is there a "smarter" way than choosing according to the nodeID?
IMO the current nvme->fc->iscsi hierarchy is good enough to allow this field to be optional.
it also allows a mixed-connectivity cluster (as currently possible with manual definition), which might be useful for some users

Reply:
about the "smart" way, I think it will take the connectivity from the NodeID, but it will know to add the connectivity that is the highest in the hierarchy and is working.

Reply:
> and is working

right, I missed that part. sounds good to me